src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinderProvider.cs

```diff
@@ -43,8 +43,6 @@ public IModelBinder GetBinder(ModelBinderProviderContext context)
             // If the model type is IEnumerable<> then we need to know if we can assign a List<> to it, since
             // that's what we would create. (The cases handled here are IEnumerable<>, IReadOnlyColection<> and
             // IReadOnlyList<>).
-            //
-            // We need to check IsReadOnly because we need to know if we can SET the property.
             var enumerableType = ClosedGenericMatcher.ExtractGenericInterface(modelType, typeof(IEnumerable<>));
             if (enumerableType != null && !context.Metadata.IsReadOnly)
```

Comment:
This should not be here anymore. Will prevent binding w/ say
``` c#
public class Model
{
    public Dictionary<string, IEnumerable<SubModel>> Property1 { get; set; }
}

public class SubModel
{
    public string Property { get; set; }
}
```
and data like `[0].Key=joe&[0].Value.Property=fred`. In this case, `SubModel` is complicated enough that the `SimpleModelBinder` won't handle it and the data matches a format `CollectionModelBinder` handles before the `DictionaryModelBinder` sees it. (Same thing goes for any `Model` with an `IEnumerable<KeyValuePair<TKey, TNotSimpleValue>>` property or a model of that ilk.)